be/server/src/main/java/com/baseball/controller/BaseballController.java

```diff
@@ -0,0 +1,62 @@
+package com.baseball.controller;
+
+import com.baseball.dto.*;
+import com.baseball.service.BaseballService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Api(tags = {"Baseball API"}, description = "야구 게임과 관련된 API")
+@RestController
+@RequestMapping("/api")
+public class BaseballController {
+
+    private final BaseballService baseballService;
+
+    public BaseballController(BaseballService baseballService) {
+        this.baseballService = baseballService;
+    }
+
+    @GetMapping("/match")
+    @ApiOperation(value = "참가가능 게임 목록 조회", notes = "참가할 수 있는 야구 게임 목록을 조회합니다.")
+    public ResponseEntity<List<MatchDto>> getMatch() {
+        List<MatchDto> matches = baseballService.getMatches();
+        return ResponseEntity.ok().body(matches);
+    }
```

Comment:
제가 공통적으로 피드백 드리는 부분인데요.
- `ResponseEntity`는 필요한 부분에서만 사용해주세요.
- `Collection` 형태의 리턴은 지양해주시고, Wrapping DTO를 생성해주셔서, Breaking Change가 덜 발생하는 API를 지향해주세요.

먼저 `ResponseEntity`를 사용하지 않은 형태는 다음과 같습니다.

```suggestion
    public List<MatchDto> getMatch() {
        return baseballService.getMatches();
    }
```

그리고, Wrapping DTO를 만들어서 리턴해야 하는 이유는 다음과 같습니다.

```json
[
 {},
 {}
]
```

이런 형태의 API는 변경에 유연하지 않습니다. 가령 목록 말고 정보를 더 주고싶다고 하면, 바로 Breaking Change가 발생합니다.

```json
{
  "matches": [],
  "matchInfo": {}
}
```

이렇게 읽기 편하고, 변경에 유연하게 만들어 줄 수 있습니다.

Reply:
오오오!!!
머리에 한방 맞은 기분입니다!!!
확실히 아래가 훨씬 유연하네요.
Collection 에 대해서는 항상 Wrapping DTO 를 만들 수 있도록 하겠습니다. 😍

Reply:
> ResponseEntity는 필요한 부분에서만 사용해주세요.

피드백 참고해서 코드를 짜는 중인데, 혹시 ResponseEntity 의 사용을 자제해야할 이유를 알 수 있을까요?
성능상에 문제가 있다던가?

ResponseEntity 를 사용했을 때는 명시적으로 HttpStatus 가 200 임을 알 수 있지만,
사용하지 않은 아래는 알기가 힘들 것 같아서요.

@ksundong

Reply:
성능상 문제라기 보다는, 가급적이면 POJO인편이 좋겠죠. ResponseEntity는 그 자체로 Spring이 녹아들어가기 때문에 저는 지양하는 편이에요.
이미 암묵적으로 `Controller`에서는 200응답을 주는 것을 우리는 알고있기 때문에 괜찮다고 생각했습니다. 😊